zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -1301,44 +1300,19 @@ private void cleanAndNotifyState() {
         private void pingRwServer() throws RWServerFoundException {
             String result = null;
             InetSocketAddress addr = hostProvider.next(0);
+            boolean useSecure = clientConfig.getBoolean(ZKClientConfig.SECURE_CLIENT);
 
             LOG.info("Checking server {} for being r/w. Timeout {}", addr, pingRwTimeout);
-
-            Socket sock = null;
-            BufferedReader br = null;
             try {
-                sock = new Socket(addr.getHostString(), addr.getPort());
-                sock.setSoLinger(false, -1);
-                sock.setSoTimeout(1000);
-                sock.setTcpNoDelay(true);
-                sock.getOutputStream().write("isro".getBytes());
-                sock.getOutputStream().flush();
-                sock.shutdownOutput();
-                br = new BufferedReader(new InputStreamReader(sock.getInputStream()));
-                result = br.readLine();
+                result = FourLetterWordMain.send4LetterWord(addr.getHostString(), addr.getPort(), "isro", useSecure, 1000);
```

Comment:
I think we should add a overloading `send4LetterWord` to accept `ZKConfig`, so we can sure that we are not mixed with external properties. Currently, `send4LetterWord` uses `new ZkConfig` which is not suitable for `ClientCnxn` after ZOOKEEPER-2139.

https://github.com/apache/zookeeper/blob/75884ec1f4a680f46bcc0a6257df413955eda4e8/zookeeper-server/src/main/java/org/apache/zookeeper/common/X509Util.java#L340-L347